patches/tools-licenses.py.patch

```diff
@@ -0,0 +1,30 @@
+diff --git a/tools/licenses.py b/tools/licenses.py
+index 791efb55b47141db7b961784f80f1dedd57e8823..4ee9eeee05c5de48f3488583717a45af02392a50 100755
+--- a/tools/licenses.py
++++ b/tools/licenses.py
+@@ -31,6 +31,8 @@ import tempfile
+ _REPOSITORY_ROOT = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
+ sys.path.insert(0, os.path.join(_REPOSITORY_ROOT, 'build/android/gyp'))
+ from util import build_utils
++sys.path.insert(0, os.path.join(_REPOSITORY_ROOT, 'brave', 'script'))
++import licenses
+ 
+ 
+ # Paths from the root of the tree to directories to skip.
+@@ -290,6 +292,7 @@ SPECIAL_CASES = {
+         "License File": "/third_party/swiftshader/LICENSE.txt",
+     },
+ }
++(PRUNE_DIRS, ADDITIONAL_PATHS) = licenses.BravePatch(PRUNE_PATHS, SPECIAL_CASES, PRUNE_DIRS, ADDITIONAL_PATHS)
+ 
+ # Special value for 'License File' field used to indicate that the license file
+ # should not be used in about:credits.
+@@ -475,7 +478,7 @@ def FindThirdPartyDirs(prune_paths, root):
+             if skip in dirs:
+                 dirs.remove(skip)
+ 
+-        if os.path.basename(path) == 'third_party':
++        if os.path.basename(path) in ('third_party', 'vendor', 'fonts'):
```

Comment:
I think we should be able to get rid of this with ADDITIONAL_PATHS

Reply:
Sadly no. I tried but it turns out that `ADDITIONAL_PATHS` is not recursive and so we'd have to list every single directory separately and we wouldn't be able to catch any omissions (once https://bugs.chromium.org/p/chromium/issues/detail?id=39240 lands).

Reply:
but by changing this aren't we we missing things that should have been added from `third_party`?

Reply:
in any case we can handle it in `ADDITIONAL_PATHS` by doing the recursion ourselves and then adding the individual entries

Reply:
> but by changing this aren't we we missing things that should have been added from `third_party`?

I'm not sure what you mean by that. `third_party` is still in the list.

> in any case we can handle it in ADDITIONAL_PATHS by doing the recursion ourselves and then adding the individual entries

So you're suggesting that we duplicate this loop in our code? I can see how that will remove one patch hunk, but it seems to me that it would be worse from a Chromium merge point of view since we'll never notice if the logic changes (and therefore we need to update our copy of it). We might have a broken `chrome://credits` page and not notice.

Reply:
brave/brave-browser#6835 will allow us to remove that part of the patch.